src/Avalonia.Visuals/Media/TextFormatting/TextFormatterImpl.cs

```diff
@@ -311,6 +327,11 @@ internal static SplitTextRunsResult SplitTextRuns(List<ShapedTextCharacters> tex
                 currentLength += textRun.Text.Length;
             }
 
+            if (textRunEnumerator.Current is TextEndOfParagraph)
```

Comment:
Remove this and introduce a default case in the switch that just adds the remaining run types to the list

Reply:
```
case TextEndOfLine endOfLine:
                    {
                        textRuns.Add(endOfLine);
                        break;
                    }
```